src/coreclr/debug/ee/debugger.cpp

```diff
@@ -16631,4 +16653,95 @@ void Debugger::StartCanaryThread()
 }
 #endif // DACCESS_COMPILE
 
+#ifndef DACCESS_COMPILE
+#if defined(TARGET_WINDOWS) && defined(TARGET_AMD64)
+void Debugger::SendSetThreadContextNeeded(CONTEXT *context)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
```

Comment:
The GC_NOTRIGGER macro calls [IncrementGCForbidCount](https://github.com/dotnet/runtime/blob/2201016c1e13bdb9abf49e2e38cadf4ee0568df2/src/coreclr/inc/contract.h#L649) which doesn't document because we SetThreadContext while inside of the method.  I don't know if it is possible to scope the contract to only part of the method.

Reply:
you can probably use STATIC_CONTRACT_GC_NOTRIGGER